test/Sema/exhaustive_switch.swift

```diff
@@ -558,3 +558,58 @@ func infinitelySized() -> Bool {
   case (.two, .two): return true
   }
 }
+
+// Literal Duplicate checks
+func checkLiterals() {
+  let str = "def"
+  let int = 2
+  let dbl = 2.5
+
+  // No Diagnostics
+  switch str {
+    case "abc": break
+    case "def": break
+    case "ghi": break
+    default: break
+  }
+
+  switch str {
+     case "abc": break
+     case "def": break // expected-note {{first occurence of identical literal pattern is here}}
+     case "def": break // expected-warning {{literal value is already handled by previous pattern; is this intentional?}}
+     case "ghi": break
+     default: break
+  }
+
+  // No Diagnostics
+  switch int {
+     case 1: break
+     case 2: break
+     case 3: break
+     default: break
+  }
+
+  switch int {
+     case 1: break
+     case 2: break // expected-note {{first occurence of identical literal pattern is here}}
+     case 2: break // expected-warning {{literal value is already handled by previous pattern; is this intentional?}}
+     case 3: break
+     default: break
+  }
+
+  // No Diagnostics
+  switch dbl {
+     case 1.5: break
+     case 2.5: break
+     case 3.5: break
+     default: break
+  }
+
+  switch dbl {
+     case 1.5: break
+     case 2.5: break // expected-note {{first occurence of identical literal pattern is here}}
+     case 2.5: break // expected-warning {{literal value is already handled by previous pattern; is this intentional?}}
+     case 3.5: break
+     default: break
```

Comment:
These tests look good, but could you add some extra duplicate cases in

- Different positions e.g.

```swift
      case 1.5: break
      case 2.5: break //
      case 2.5: break //
      case 3.5: break
      case 2.5: break //
      default: break
```

- Different levels of normalization:

```swift
      case 1.5: break
      case 1.5000: break
      case 1.500: break

///
      case 1
      case 0b1
      case 0x1
```